### DISCUSSION TITLE
Canvas object property is not updating; used useState (Fabricjs with react)

### DISCUSSION THREAD

**Discussion by rohitkanojiya** [fabricjs/fabric.js#7855] [Q&A] [answered]
Hello, 

I am using FabricJS with React, and I am able to add objects easily to the canvas but facing problems when trying to manipulate objects' properties, For example, I have recreated my issue in JSfiddle link: https://jsfiddle.net/rohitkanojiya/Lh51ops2/24/

where I am trying to toggle draw by changing canvas object property isDrawingMode to true and false.

Please help, 
Thanks.

**Comment by asturur** [comment, marked as answer]
When working with react you have to divide events in 2 categories.
Things that the user can do on his own without using your code ( dragging, scaling, rotating ) and things that are triggered by your code.
When things are triggered by your code, you are in charge and you need to update fabric and react.
Seems a bad idea to use a useState, to generate a render, to reinitizalize the canvas, to call another setState.


```
  const toggleDraw = () => {
    if(canvas.isDrawingMode){
      setDraw(false)
      canvas.isDrawingMode = false;
      return
    }
    if(!canvas.isDrawingMode){
      setDraw(true)
      canvas.isDrawingMode = true;
      return
    }

  }
```

I would do something like this to keep it simple.
But going forward and implementing different functionalities things get a little be more complicated

**Comment by rohitkanojiya** [comment]
Hi @asturur, Thanks for the response.
It's working, and i understood that I will need to manipulate direct properties of canvas object and render it again. no need to use  useState for this simple changes.

Thanks again :)
